src/app/portlets/dot-edit-page/main/dot-edit-page-nav/dot-edit-page-nav.component.ts

```diff
@@ -31,13 +32,18 @@ export class DotEditPageNavComponent implements OnChanges {
     constructor(private dotLicenseService: DotLicenseService, public dotMessageService: DotMessageService, public route: ActivatedRoute) {}
 
     ngOnChanges(changes: SimpleChanges): void {
-        if (changes.pageState.currentValue !== changes.pageState.previousValue) {
+        if (this.layoutChanged(changes)) {
             this.model = !this.model ? this.loadNavItems() : Observable.of(this.getNavItems(this.pageState, this.isEnterpriseLicense));
         }
     }
 
-    private loadNavItems(): Observable<DotEditPageNavItem[]> {
+    private layoutChanged(changes: SimpleChanges): boolean {
```

Comment:
this method signature should be 

 private layoutChanged(oldPageState: DotRenderedPageState, newPageState: DotRenderedPageState): boolean {

Reply:
mmm, why?

Reply:
inside the method it's used `changes.pageState.firstChange`, so according to your comment we should also add that third parameter :-S